java/com/jfreechartextensions/renderers/ExtendedXYStepRenderer.java

```diff
@@ -120,9 +122,8 @@ public void drawItem(Graphics2D g2, XYItemRendererState state,
                     }
 
                     // submit this data item as a candidate for the crosshair point
-                    int domainAxisIndex = plot.getDomainAxisIndex(domainAxis);
-                    int rangeAxisIndex = plot.getRangeAxisIndex(rangeAxis);
-                    updateCrosshairValues(crosshairState, x1, y1, domainAxisIndex, rangeAxisIndex, transX1, transY1, orientation);
+                    int datasetIndex = plot.indexOf(dataset);
```

Comment:
Do we want to have state.getDataset() instead of dataset.
I don't have any idea, just want to draw your attention.

Reply:
Putting the response here: the dataset used in the drawItem() is the dataset from the state